proto/src/test/java/com/bakdata/kafka/ProtoDeadLetterTransformerTest.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.bakdata.kafka;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.bakdata.kafka.proto.v1.Deadletter.DeadLetter;
+import com.google.protobuf.Int32Value;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig;
+import io.confluent.kafka.streams.serdes.protobuf.KafkaProtobufSerde;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+import org.apache.kafka.streams.kstream.Produced;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.jooq.lambda.Seq;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(SoftAssertionsExtension.class)
+class ProtoDeadLetterTransformerTest extends ErrorCaptureTopologyTest {
+    @InjectSoftAssertions
+    private SoftAssertions softly;
+    private static final String ERROR_TOPIC = "errors";
+    private static final String OUTPUT_TOPIC = "output";
+    private static final String INPUT_TOPIC = "input";
+    private static final Serde<String> STRING_SERDE = Serdes.String();
+    public static final String DEADLETTER_DESCRIPTION = "Description";
+    public static final String ERROR_MESSAGE = "ERROR!";
+
+    private Serde<DeadLetter> configuredDeadLetterSerde = null;
+    @Mock
+    KeyValueMapper<Integer, String, KeyValue<Integer, String>> mapper;
+
+    Serde<DeadLetter> deadLetterSerde() {
+        if (this.configuredDeadLetterSerde == null) {
+            this.configuredDeadLetterSerde = new KafkaProtobufSerde<>(DeadLetter.class);
+            this.configuredDeadLetterSerde.configure(Map.of(
+                    AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, "mock://test"), false);
```

Comment:
why not use the SR mock?

Reply:
The topology and the SR mock are not yet available when building the topology. Do you see a non-ugly workaround?

Reply:
Well here it seems to work: https://github.com/bakdata/kafka-error-handling/pull/13/files#diff-74f44e129bb101e8fd8c90cb9449e18f7163e6f49a6fb64593639d574ffcf678